File: pytools/data/data.code.lua.server/ReqSceneMove.lua
module("ReqSceneMove", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqSceneMove})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.scene_rot_pos = nil
	self.forward = nil
	self.ani_name = 0
	self.x_axis = 0
end


function decode(self, pack)
	self.scene_rot_pos = MsgSceneRotPos.new():decode(pack)
	self.forward = MsgSceneVector3.new():decode(pack)
	self.ani_name = pack:readString()
	self.x_axis = pack:readI16()
	return self
end


function getSceneRotPos(self)
	return self.scene_rot_pos
end

function getForward(self)
	return self.forward
end

function getAniName(self)
	return self.ani_name
end

function getXAxis(self)
	return self.x_axis
end

File: pytools/data/data.code.php.server/ReqSceneMove.php
<?php
namespace protocol;

use network\Packet;


class ReqSceneMove
{
	private $sceneRotPos = null;
	private $forward = null;
	private $aniName = '';
	private $xAxis = 0;


	public function __construct(&$packet)
	{
		$this->decode($packet);
	}

	public function decode(&$packet)
	{
		$this->sceneRotPos = new MsgSceneRotPos($packet);
		$this->forward = new MsgSceneVector3($packet);
		$this->aniName = $packet->readString();
		$this->xAxis = $packet->readI16();
	}


	public function getSceneRotPos()
	{
		return $this->sceneRotPos;
	}

	public function getForward()
	{
		return $this->forward;
	}

	public function getAniName()
	{
		return $this->aniName;
	}

	public function getXAxis()
	{
		return $this->xAxis;
	}

}
